### ISSUE TITLE
Add Flux.jl demo

### ISSUE THREAD

**Issue by oxinabox** [PhilipVinc/TensorBoardLogger.jl#37]
Maybe by cleaning up the following:

```
flat(xs) = reduce(vcat, vec.(data.(xs)))

function train!(loss_func, mdl, features, outputs, settings)
    training_record = NamedTuple{(:loss,),Tuple{Float32}}[]

    network_parameters = params(mdl)
    @assert length(network_parameters) > 0

    if LOG_TENSORBOARD
        lg = TBLogger("logs/run"; min_level=Logging.Debug)
    else
        lg = current_logger()
    end
    with_logger(lg) do
        for iter in 1:settings.max_epochs
            grad = gradient(network_parameters) do
                preds = mdl(features)
                training_loss = loss_func(preds, outputs, settings)
                push!(training_record, (loss=data(training_loss),))

                @debug "" network_parameters = flat(network_parameters)
                @debug "" generated_volumes = vec(data(volumes))
                @debug "" training_loss = data(training_loss)

                iter += 1
                return training_loss
            end
            @debug "" mean_grad=mean(abs, flat(values(grad.grads)))
            update!(settings.optimizer, network_parameters, grad)
            last_loss = training_record[end].loss
        end
    end
    return training_record
end
```

**Comment by oxinabox** [comment]
see https://gist.github.com/oxinabox/f10edd1bda4fb9ebd8e06979fe37b960

**Comment by PhilipVinc** [comment]
Can you even run this?
My computer has spent the last hour running 
`X = hcat(float.(reshape.(imgs, :))...)`

I'll try on a cluster with a Gpu, too, but maybe something that takes that long to even run is not the best example...

**Comment by oxinabox** [comment]
this is a sketch, simplified from some other code.
it may have wierdness.

**Comment by PhilipVinc** [comment]
Closed by #60
